### ISSUE TITLE
Support DDIC domain append values (APPENDNAME) in check_ddic/unused_ddic

### ISSUE THREAD

**Issue by IGNVDRON** [abaplint/abaplint#4212]
### Description

`abaplint` treats DDIC domain append values exported by abapGit as regular domains.
 
For example - custom append ZDEVCLASS with value 'DEVCLASS' for standard domain MENU_ATT3 (XML is produced by abapGit):
```xml
<?xml version="1.0" encoding="utf-8"?>
<abapGit version="v1.0.0" serializer="LCL_OBJECT_DOMA" serializer_version="v1.0.0">
 <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
  <asx:values>
   <DD01V>
    <DOMNAME>ZDEVCLASS</DOMNAME>
    <DDLANGUAGE>E</DDLANGUAGE>
    <VALEXI>X</VALEXI>
    <DDTEXT>Package</DDTEXT>
    <APPENDNAME>MENU_ATT3</APPENDNAME>
   </DD01V>
   <DD07V_TAB>
    <DD07V>
     <VALPOS>0001</VALPOS>
     <DDLANGUAGE>E</DDLANGUAGE>
     <DOMVALUE_L>DEVCLASS</DOMVALUE_L>
     <DDTEXT>Package</DDTEXT>
    </DD07V>
   </DD07V_TAB>
  </asx:values>
 </asx:abap>
</abapGit>
```

This object is not a standalone domain. It is an append containing additional fixed values for the existing domain `MENU_ATT3`.

### Current behavior
`check_ddic` reports "_Unknown/un-resolveable type in ZDEVCLASS: undefined unknown basic ddic type_" because the append object intentionally has no datatype/length.
`unused_ddic` reports "_DOMA ZDEVCLASS not statically referenced_" although it is referenced by `MENU_ATT3` in `<APPENDNAME>` and is not expected to be used directly in ABAP code.
<img width="1358" height="798" alt="Image" src="https://github.com/user-attachments/assets/243f1261-da31-4716-aa6a-317648bf4600" />

### Expected behavior
If a domain XML contains `<APPENDNAME>`, `abaplint` should recognize it as a domain append:
- `check_ddic` should not expect datatype metadata.
- `unused_ddic` should not report it as an unused standalone domain.

### Workaround
Append domains can be added to `exclude[ ]` for `check_ddic` and `unused_ddic`.
